pkg/process/init.go

```diff
@@ -173,6 +181,57 @@ func (p *Init) Create(ctx context.Context, r *CreateConfig) error {
 	return nil
 }
 
+func (p *Init) validateRuncFeatures(ctx context.Context, bundle string) error {
+	spec, err := readConfig(bundle)
+	if err != nil {
+		return fmt.Errorf("failed to read config: %w", err)
+	}
+	if spec == nil {
+		return errors.New("config.json is empty")
+	}
+
+	if err := p.validateIDMapMounts(ctx, spec); err != nil {
+		return fmt.Errorf("OCI runtime doesn't support idmap mounts: %w", err)
+	}
+
+	return nil
+}
+
+func (p *Init) validateIDMapMounts(ctx context.Context, spec *specs.Spec) error {
+	var used bool
+	for _, m := range spec.Mounts {
+		if m.UIDMappings != nil || m.GIDMappings != nil {
+			used = true
+			break
+		}
+	}
+
+	if !used {
+		return nil
+	}
+
+	// From here onwards, we require idmap mounts. So if we fail to check, we return an error.
+	features, err := p.runtime.Features(ctx)
+	if err != nil {
+		// If the features command is not implemented, then runc is too old.
+		return fmt.Errorf("features command failed: %w", err)
+
+	}
+	if features == nil {
```

Comment:
It won't be nil. In go convension, if err is nil, the result should be not nil.

Reply:
This can technically happen if a runtime produces a null output. This shouldn't happen, but crashing containerd here seems very nasty.

From https://pkg.go.dev/encoding/json#Unmarshal:

> To unmarshal JSON into a pointer, Unmarshal first handles the case of the JSON being the JSON literal null. In that case, Unmarshal sets the pointer to nil.

I'd prefer to leave it, just in case. It is very simple, only a few lines, and doesn't make the code unreadable nor anything, it is just here. What do you think @fuweid ?

Reply:
I think the API writer should ensure that resp should not be nil. If it's panic, we should fix it as soon as possible.
Fortunately, the implementation ensures it won't be nil if the err is nil. :) 

https://github.com/containerd/go-runc/blob/df3c22d088ca919d1943efa8ae0cfc0514da0542/runc.go#L761

Reply:
Oh, cool. Removed that too, then! :)